variables.tf

```diff
@@ -208,3 +208,9 @@ variable "allow_version_upgrade" {
   type        = bool
   default     = true
 }
+
+variable "max_concurrency_scaling_clusters" {
+  description = "(Optional) Max concurrency scaling clusters parameter (0 to 10)"
+  type        = string
+  default     = "1"
```

Comment:
Is `1` a good default? Isn't it better to set to `null` (and update code to exclude this parameter if it is `null`) to keep backward compatibility?

Reply:
Hi @antonbabenko 
Yes, I think `1` is good default value for 2 reasons:
1. If parameter is not declared in custom parameter group, it inherited from default parameter group, where its value is `1`. 
So applying `max_concurrency_scaling_clusters` with default value of `1` changes parameter source only. Example below (note that this data only visible in CLI, not AWS console. [Link](https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html)):

Before terraform apply
```bash
$ aws redshift describe-cluster-parameters --parameter-group-name custom-parameter-group-name
...
{
            "ParameterName": "max_concurrency_scaling_clusters",
            "ParameterValue": "1",
            "Description": "The maximum concurrency scaling clusters can be used.",
            "Source": "engine-default",
            "DataType": "integer",
            "AllowedValues": "0-10",
            "ApplyType": "static",
            "IsModifiable": true
},
...
```

After terraform apply:
```bash
$ aws redshift describe-cluster-parameters --parameter-group-name custom-parameter-group-name
...
{
            "ParameterName": "max_concurrency_scaling_clusters",
            "ParameterValue": "1",
            "Description": "The maximum concurrency scaling clusters can be used.",
            "Source": "user",
            "DataType": "integer",
            "AllowedValues": "0-10",
            "ApplyType": "static",
            "IsModifiable": true
},
...
```
Note the only difference is `Source` value. That doesn't implies any changes to the cluster.

2. And second reason is that `parameter` block in [redshift_parameter_group](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/redshift_parameter_group#parameter) doesn't support conditions, so I don't think we can exclude it in case of `null` or empty variable.